sdk/servicebus/azservicebus/client.go

```diff
@@ -0,0 +1,103 @@
+package azservicebus
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/servicebus/azservicebus/internal"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/devigned/tab"
+)
+
+type ServiceBusClient struct {
+	namespace *internal.Namespace
+	linksMu   *sync.Mutex
+	links     []interface {
+		Close(ctx context.Context) error
+	}
+}
+
+// ServiceBusClientOption is the type for an option that can configure ServiceBusClient.
+// For an example option, see `ServiceBusWithConnectionString`
+type ServiceBusClientOption func(client *ServiceBusClient) error
```

Comment:
While I would prefer to use the variadic config pattern, it would be different than our other track 2 SDKs as well as our design guidelines.  I think we need to have a discussion about where we want to land on this.

Reply:
Is the alternative just taking a `struct` as a parameter? I would have loved to do that (although the queue vs topic+subscription) might be a bit painful.

Who do we kick this discussion off with?

Reply:
Yeah a struct.  https://azure.github.io/azure-sdk/golang_introduction.html#optional-parameters

We really need @JeffreyRichter to be here for this discussion.  You can leave it as-is for now, but this might change depending on how the conversation goes.

Reply:
Sounds good.